pyvista/plotting/colors.py

```diff
@@ -344,17 +348,506 @@
+            raise ValueError(
+                "\n"
+                f"\tInvalid opacity input: ({opacity})"
+                "\tMust be an integer, float or string.  For example:\n"
+                "\t\topacity='1.0'\n"
+                "\t\topacity='255'\n"
+                "\t\topacity='#FF'"
+            ) from e
+
+    @staticmethod
+    def strip_hex_prefix(h: str) -> str:
+        """Strip any ``'#'`` or ``'0x'`` prefix from a hexadecimal string.
+
+        Parameters
+        ----------
+        h : str
+            Hexadecimal string to strip.
+
+        Returns
+        -------
+        str
+            Stripped hexadecimal string.
+
+        """
+        h = h.lstrip('#')
+        if h.startswith('0x'):
+            h = h[2:]
+        return h
+
+    @staticmethod
+    def convert_color_channel(val: Union[int, np.integer, float, np.floating, str]) -> int:
+        """Convert the given color channel value to the integer representation.
+
+        Parameters
+        ----------
+        val : int, float or str
+            Color channel value to convert. Supported input values are a
+            hex string of length 2 (``'00'`` to ``'ff'``) with an optional
+            prefix (``'#'`` or ``'0x'``), a float (``0.0`` to ``1.0``) or
+            an integer (``0`` to ``255``).
+
+        Returns
+        -------
+        int
+            Color channel value in the integer representation (values between
+            ``0`` and ``255``).
+
+        """
+        if isinstance(val, str):
+            # From hexadecimal value
+            val = int(Color.strip_hex_prefix(val), 16)
+        elif np.issubdtype(np.asarray(val).dtype, np.floating) and np.ndim(val) == 0:
+            # From float
+            val = int(round(255 * val))
+        if (
+            np.issubdtype(np.asarray(val).dtype, np.integer)
+            and np.size(val) == 1
+            and 0 <= val <= 255
+        ):
+            # From integer
+            return int(val)
+        else:
+            raise ValueError(f"Unsupported color channel value provided: {val}")
+
+    def _from_rgba(self, rgba):
+        """Construct color from an RGB(A) sequence."""
+        arg = rgba
+        if len(rgba) == 3:
+            # Keep using current opacity if it is not provided.
+            rgba = [*rgba, self._opacity]
+        try:
+            if len(rgba) != 4:
+                raise ValueError("Invalid length for RGBA sequence.")
+            self._red, self._green, self._blue, self._opacity = [
+                self.convert_color_channel(c) for c in rgba
+            ]
+        except ValueError:
+            raise ValueError(f"Invalid RGB(A) sequence: {arg}")
+
+    def _from_dict(self, dct):
+        """Construct color from an RGB(A) dictionary."""
+        r = dct.get("r", dct.get("red", None))
+        g = dct.get("g", dct.get("green", None))
+        b = dct.get("b", dct.get("blue", None))
+        a = dct.get("a", dct.get("alpha", dct.get("opacity", None)))
```

Comment:
Another nitpick: it might be more future-proof to do something like
```py
r = next((v for k, v in dct.items() if k in {'r', 'red'}), None)
```
Then if we need to add further synonyms we don't have to add more nesting.